File: python/leetcode/252.py
class Solution:
    def canAttendMeetings(intervals):
        intervals.sort(key = lambda x: x.start)
        end = intervals[0].end
        for i in range(1, len(intervals)):
            if intervals[i].start < end: return False
            end = max(end, intervals[i].end)
        return True 

File: cpp/leetcode/252.cpp
class Solution {
public:
    bool canAttendMeetings(vector<Interval>& intervals) {
        sort(intervals.begin(), intervals.end(), compare);
        for(int i = 1; i < intervals.size(); i++){
            if(intervals[i-1].end > intervals[i].start) return false;
        }
        return true;
    }

private:
    static bool compare(Interval a, Interval b){
        return a.start < b.start;
    }
};


class Solution {
public:
    bool canAttendMeetings(vector<Interval>& intervals) {
        sort(intervals.begin(), intervals.end(), [](Interval a, Interval b){return a.start < b.start;});
        for(int i = 1; i < intervals.size(); i++){
            if(intervals[i-1].end > intervals[i].start) return false;
        }
        return true;
    }
};